srml/contract/src/rent.rs

```diff
@@ -37,11 +37,12 @@ pub enum RentOutcome {
 	Paid,
 }
 
-/// Evict and optionally pay dues (or check account can pay them otherwise), for given block number.
-/// Return true iff the account has been evicted.
+/// Evict and optionally pay dues (or check account can pay them otherwise) at the current
+/// block number (modulo `handicap`, read on).
 ///
-/// `pay_rent` gives an ability to pay or skip paying rent. This is useful for the
-/// `try_evict_at` use-case.
+/// `pay_rent` gives an ability to pay or skip paying rent.
+/// `handicap` gives a way to shift the state of the contract to some moment in the past.
```

Comment:
it doesn't shift the state of the contract to some moment in the past. For example the balance of the contract is the current balance not the one at the moment in the past.

I would write this way: gives a way to check_or_pay the rent up to a moment in the past instead of current block